pennylane/ops/qubit/parametric_ops_single_qubit.py

```diff
@@ -613,18 +613,18 @@ def compute_matrix(phi, theta, omega):  # pylint: disable=arguments-differ
         c = c * one
         s = s * one
 
-        mat = [
-            [
-                qml.math.exp(-0.5j * (phi + omega)) * c,
-                -qml.math.exp(0.5j * (phi - omega)) * s,
-            ],
+        pp = qml.math.exp(0.5j * (phi + omega))
+        pm = qml.math.exp(0.5j * (phi - omega))
+        if hasattr(one, "reshape"):
+            return stack_last(
+                [qml.math.conj(pp) * c, -pm * s, qml.math.conj(pm) * s, pp * c]
+            ).reshape((*c.shape, 2, 2))
+        return stack_last(
```

Comment:
@albi3ro Would you like me to remove these changes before merging? This one, unlike `operation.py` is not overlapping with #4261 I think.